sdk/communication/communication-sms/test/smsClient.mocked.spec.ts

```diff
@@ -45,95 +46,88 @@ describe("[mocked] SmsClient", async () => {
       };
     }
   };
-
   const mockedGuid = "42bf408f-1931-4314-8971-2b538625a2b0";
 
-  beforeEach(() => {
-    sinon.stub(Uuid, "generateUuid").returns(mockedGuid);
-  });
+  const testSendRequest: SmsSendRequest = {
+    from: "+15558984505651",
+    to: ["+15558985487"],
+    message: "message"
+  };
+  let testResults: SmsSendResult[];
+
+  //Helper functions
+  const shouldReturnCorrectValues = () => {
+    it("returns the correct results", () => {
+      const _response = testResults[0];
+      sinon.assert.calledOnce(spy);
+      assert.equal(_response.httpStatusCode, 202);
+      assert.equal(_response.messageId, "id");
+    });
+  };
 
-  afterEach(() => {
-    sinon.restore();
-  });
+  const shouldSendCorrectRequest = () => {
+    it("sends with the correct request body", () => {
+      const request = spy.getCall(0).args[0];
+      assert.equal(request.url, `${baseUri}/sms?api-version=${API_VERSION}`);
+      assert.equal(request.method, "POST");
+      const expectedRequestBody = generateSendMessageRequest(testSendRequest);
+      assert.deepEqual(JSON.parse(request.body), expectedRequestBody);
+    });
+  };
 
-  it("can instantiate with a connection string", async () => {
-    new SmsClient("endpoint=https://contoso.communicationservices.azure.com:443/;accesskey=secret");
-  });
+  const shouldSetCorrectHeaders = () => {
+    it("sends with the correct headers", () => {
+      const request = spy.getCall(0).args[0];
+      if (isNode) {
+        assert.equal(request.headers.get("host"), "contoso.api.fake:443");
+      }
+      assert.typeOf(request.headers.get(dateHeader), "string");
+      assert.isDefined(request.headers.get("authorization"));
+      assert.match(
+        request.headers.get("authorization") as string,
+        /HMAC-SHA256 SignedHeaders=.+&Signature=.+/
+      );
+    });
+  };
 
-  it("sends a SMS when url and KeyCredential are provided", async () => {
-    smsClient = new SmsClient(baseUri, new AzureKeyCredential("banana"), {
-      httpClient: mockHttpClient
+  describe("when client instantiated with connection string", async () => {
+    const connectionString = `endpoint=${baseUri};accesskey=banana`;
+    before(async () => {
+      sinon.stub(Uuid, "generateUuid").returns(mockedGuid);
+      spy = sinon.spy(mockHttpClient, "sendRequest");
+      sinon.useFakeTimers();
+
+      smsClient = new SmsClient(connectionString, { httpClient: mockHttpClient });
+      testResults = await smsClient.send(testSendRequest);
```

Comment:
looking to solicit feedback if it's ok to perform the send in the "before" portion, or if I should actually do it in each helper test case

Reply:
decided to take it out of here, and instead run send for each test case. It's a mock anyway